android/src/main/java/com/mattermost/networkclient/APIClientModule.kt

```diff
@@ -101,15 +101,22 @@ class APIClientModule(reactContext: ReactApplicationContext) : ReactContextBaseJ
     private fun parseOptions(baseUrl: String, options: ReadableMap): ReadableMap {
         val timeoutInterval = options.getInt("timeoutInterval")
         if(timeoutInterval > 0){
-            val duration = timeoutInterval.toLong()
-            sessionsClient[baseUrl]!!.readTimeout(duration, TimeUnit.SECONDS)
+            sessionsClient[baseUrl]!!.addNetworkInterceptor(TimeoutInterceptor(timeoutInterval))
         }
 
         return options;
     }
 
+    private fun parseOptionsClean(baseUrl: String, options: ReadableMap) {
```

Comment:
Yup I think renaming it is probably better; I'm hesitant to modify the session in the methods because we then need to add each config option to each method - that's a lot of duplicated code across the get/post/put/patch/delete - and will be a bit of a pain to update; especially if we add/remote/update a given config option.

I think in psuedo-code it's the difference between:

```
fun get(...){
  modifySessionOnOptions(options);
  ... execute request;
  cleanSessionAsNeeded(options);
}
fun post(...){
  modifySessionOnOptions(options);
  ... execute request;
  cleanSessionAsNeeded(options);
}
...
```

vs

```
fun get(...){
  if(checkForOptionA(options)) doOptionA();
  if(checkForOptionB(options)) doOptionB();
  ... execute request;
   if(checkForOptionA(options)) cleanOptionA();
  if(checkForOptionB(options)) cleanOptionB();
}
fun post(...){
  if(checkForOptionA(options)) doOptionA();
  if(checkForOptionB(options)) doOptionB();
  ... execute request;
   if(checkForOptionA(options)) cleanOptionA();
   if(checkForOptionB(options)) cleanOptionB();
}
```

Pretty sure the potential config options we'll be passing to each method (and even the session initialiser) will be quite a lot eventually.